src/cascadia/TerminalApp/JsonUtilsNew.h

```diff
@@ -0,0 +1,437 @@
+        }
+    };
+
+    template<>
+    struct ConversionTrait<GUID>
+    {
+        GUID FromJson(const Json::Value& json)
+        {
+            return ::Microsoft::Console::Utils::GuidFromString(til::u8u16(Detail::GetStringView(json)));
+        }
+
+        bool CanConvert(const Json::Value& json)
+        {
+            if (!json.isString())
+            {
+                return false;
+            }
+
+            const auto string{ Detail::GetStringView(json) };
+            return string.length() == 38 && string.front() == '{' && string.back() == '}';
+        }
+    };
+
+    // (GUID and winrt::guid are mutually convertible!)
+    template<>
+    struct ConversionTrait<winrt::guid> : public ConversionTrait<GUID>
+    {
+    };
+
+    template<>
+    struct ConversionTrait<til::color>
+    {
+        til::color FromJson(const Json::Value& json)
+        {
+            return ::Microsoft::Console::Utils::ColorFromHexString(Detail::GetStringView(json));
+        }
+
+        bool CanConvert(const Json::Value& json)
+        {
+            if (!json.isString())
+            {
+                return false;
+            }
+
+            const auto string{ Detail::GetStringView(json) };
+            return (string.length() == 7 || string.length() == 3) && string.front() == '#';
+        }
+    };
+
+    template<typename T, typename TBase>
+    struct EnumMapper
+    {
+        using pair_type = std::pair<std::string_view, T>;
+        T FromJson(const Json::Value& json)
+        {
+            const auto name{ Detail::GetStringView(json) };
+            for (const auto& pair : TBase::mappings)
+            {
+                if (pair.first == name)
+                {
+                    return pair.second;
+                }
+            }
+
+            throw std::exception{};
+        }
+
+        bool CanConvert(const Json::Value& json)
+        {
+            return json.isString();
+        }
+    };
+
+    // FlagMapper is EnumMapper, but it works for bitfields.
+    // It supports a string (single flag) or an array of strings.
+    // Does an O(n*m) search; meant for small search spaces!
+    //
+    // Cleverly leverage EnumMapper to do the heavy lifting.
+    template<typename T, typename TBase>
+    struct FlagMapper : public EnumMapper<T, TBase>
+    {
+        static constexpr T AllSet{ static_cast<T>(~0u) };
+        static constexpr T AllClear{ static_cast<T>(0u) };
+
+        T FromJson(const Json::Value& json)
+        {
+            if (json.isString())
+            {
+                return EnumMapper::FromJson(json);
+            }
+            else if (json.isArray())
+            {
+                unsigned int seen{ 0 };
+                T value{};
+                for (const auto& element : json)
+                {
+                    const auto newFlag{ EnumMapper::FromJson(element) };
+                    if (++seen > 1 &&
+                        ((newFlag == AllClear && value != AllClear) ||

```

Comment:
NOTE: I tried `(value|newFlag) == value`, but it makes `"all", "first"` _invalid_. I don't see a compelling reason to make that invalid, whereas for `"none", "first"` I do.